reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientOperations.java

```diff
@@ -284,6 +285,11 @@ protected void onInboundCancel() {
 		channel().close();
 	}
 
+	@Override
+	protected final void onUnprocessedOutboundError(Throwable t) {
+		this.unprocessedOutboundError = t;
```

Comment:
Are we sure that `reactor.netty.channel.ChannelOperations#onError` is invoked in an event loop? Let say the client sends something like this

```
.send((req, out) -> Mono.error(new RuntimeException("something")).subscribeOn(Schedulers.boundedElastic()).then())
```

Reply:
thanks, indeed, in this case, the onError callback can be definitely invoked from any thread, and not only from event loops.
I have set the field as volatile for safety, but I have difficulties to try to improve the test in order to reproduce a scenario where we have:
- the onError invoked from a separate non-event loop thread (by something similar to the example you provided)
- **AND** the isDisposed() method from the ChannelOperations.onError method is returning true
- **AND** ChannelOperations.onInboundClose() is called with both isInboundCancelled() and isInboundDisposed() methods returning false , so the exceptions are enriched with the unprocessedOutboundError

do you think I should try to insist on doing a test which demonstrates the concurrency issue in case the field is not volatile ? I'm confused for the moment.

Reply:
`volatile` is ok, if a test is hard to be achieved then let's skip it for the moment.